streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorder.java

```diff
@@ -462,8 +462,10 @@ public void record(final long now) {
             writeStallDuration += valueProviders.statistics.getAndResetTickerCount(TickerType.STALL_MICROS);
             bytesWrittenDuringCompaction += valueProviders.statistics.getAndResetTickerCount(TickerType.COMPACT_WRITE_BYTES);
             bytesReadDuringCompaction += valueProviders.statistics.getAndResetTickerCount(TickerType.COMPACT_READ_BYTES);
-            numberOfOpenFiles += valueProviders.statistics.getAndResetTickerCount(TickerType.NO_FILE_OPENS)
-                - valueProviders.statistics.getAndResetTickerCount(TickerType.NO_FILE_CLOSES);
+
+            // NO_FILE_CLOSES metric was removed in RocksDB 9.7.3.
+            // This now tracks the total number of file opens since the last reset.
+            numberOfOpenFiles += valueProviders.statistics.getAndResetTickerCount(TickerType.NO_FILE_OPENS);
```

Comment:
As commented on the Jira ticket, changing semantics of an existing metric does not sound right. I would rather just report `-1` surrogate for now, and deprecate this metric with 4.0 release.

Curious to hear what others think.

Reply:
Tend to agree that changing semantics is no good, but it sounds like the new metric will be useful -- is it possible to deprecate the old one and introduce a new one that corresponds to the same underlying thing? I really don't know anything about how we expose the RocksDB metrics though so Bruno is probably better suited to chime in

Reply:
@mjsax thanks for feedback. Can you tell me how you want me to do that? As NO_FILE_CLOSES metric declared inside TickerType class in org.rocksdb package.